src/server.c

```diff
@@ -5671,6 +5704,14 @@ sds genRedisInfoString(dict *section_dict, int all_sections, int everything) {
             "total_watched_keys:%lu\r\n", watched_keys,
             "total_blocking_keys:%lu\r\n", blocking_keys,
             "total_blocking_keys_on_nokey:%lu\r\n", blocking_keys_on_nokey));
+
+        if (server.io_threads_num > 1) {
+            info = sdscatprintf(info, "main_thread_clients:%ld\r\n",
+                server.io_threads_clients_num[IOTHREAD_MAIN_THREAD_ID] - listLength(server.slaves));
+            for (j = 1; j < server.io_threads_num; j++) {
+                info = sdscatprintf(info, "io_thread_%d_clients:%d\r\n", j, server.io_threads_clients_num[j]);
```

Comment:
maybe we should rename this to `io_thread_clients%d`
or change to a dict like
`thread%d:clients=%d` like keyspace and slaves info.
some metric scraping systems take metrics that end with a number to convert them to dicts like `slave0` to `slave` with `idx=0` attribute.
maybe instead main_thread_clients it can be `thread0`?

Reply:
ohh, i realize that my last suggestion (`thread%d:clients=%d`) is odd considering this is in the `clients` section.
if we do that, maybe it should be in another section?

Reply:
```
io_thread_clients0:
io_thread_clients1:
```
looks not clear. `clients0`, `clients1`  are confusing, not as clear as `clients_io_thread_1`, even it breaks the style of ending with the word `clients`.


If we want to use dict like `thread%d:clients=%d`, i prefer to add a new section, since here is `clients` section, we should not add more fields for io threads.
```
# Threads
io_thread_0:clients=0
io_thread_1:clients=1
```

Reply:
i agree the first option is odd, and the second option is misplaced.
out of all these options, i'd prefer to add a section.
but i'm not sure who approves such interface changes today.

Reply:
If we add a new section, one thing we also should think is that these fields of threads may should belong to other sections, it is not as independent as `slave0` or `db0`. For example, `clients` field is more suitable in clients section, otherwise, we may want to show cpu usages of IO threads, it is better to show them in CPU section, since in CPU section, we have main thread info.
```
# CPU
used_cpu_sys:456.662508
used_cpu_user:271.181048
used_cpu_sys_children:0.000000
used_cpu_user_children:0.000000
used_cpu_sys_main_thread:8.191489
used_cpu_user_main_thread:195.654534
```
if we add IO thread CPU info, we add them at the end, as below
```
used_cpu_sys:456.662508
used_cpu_user:271.181048
used_cpu_sys_children:0.000000
used_cpu_user_children:0.000000
used_cpu_sys_main_thread:8.191489
used_cpu_user_main_thread:195.654534
used_cpu_sys_io_thread_1:3.14
used_cpu_user_io_thread_1:0.618
used_cpu_sys_io_thread_2:3.14
used_cpu_user_io_thread_2:0.618
```

Let's take a look at the clients section again，now we have `clients_in_timeout_table` in clients section, so for io threads clients info, maybe the following is another option.
```
clients_in_io_thread_1: 12
clients_in_io_thread_2: 12
```

about the `_` between io_thread and 1, if it doesn't affect metric scraping systems taking, i prefer `io_thread_1`, since the thread title just uses this style `io_thd_1`, `io_thd_2`, if it does affect taking, `io_thread1` and `iothread1` both are ok, maybe `iothread1` is better.

Reply:
i think we can create a new section for `threads`, and add lines like `io_thread_0:clients=0` which will include other per-thread info later on.
the `clients` section can have total (total number of clients), the `server` section can have total CPU utilization, and the `threads` section can have per-thread breakdown (kinda like the CLIENT LIST, or INFO COMMANDSTATS that offer per client or per command info).

i'm also ok with just `clients_in_io_thread_1` in the `clients` section.

let's pull @YaacovHazan for another opinion.

p.s. let's not forget to consider renaming the `main` thread to `thread_0` if it makes sense (depends on which of the above we'll chose)

Reply:
for renaming  the `main` thread to `thread_0`
In the beginning, i just did that, but i notice we have already used `used_cpu_sys_main_thread`  in `CPU` sections

Reply:
i suppose that when we have "main" and "others" or "io" it's the right thing to do.
but if we have some generic per-thread info (especially if we add the "threads" section), maybe it's better to just give it a number.
if we need, we can also add a per-thread attribute like `thread_0:clients=1,type=io` and `thread_0:clients=0,type=main`.
it's kinda hard to foresee what we'll want to do with that in the future, so it's hard to make a call.

Reply:
@oranagra If we think a new section is the right choice, we can add it.

Reply:
i was hoping you'll choose.
well, it's hard to know what the future will bring, and any choice we make can later turn out to be the wrong one.
i'll let @ShooterIT to chose based on the discussion above.
the only thing i'll ask is that the thread index is at the end of the info field name.

Reply:
@YaacovHazan @oranagra Added a new section named `Threads` [b53c80c](https://github.com/redis/redis/pull/13665/commits/b53c80caa1d30c8aa187ecdf9d3053faed882141).

the output is as follows

multi IO threads:
```
# Threads
io_thread_0:clients=0
io_thread_1:clients=2
io_thread_2:clients=2
```
single IO thread:
```
# Threads
io_thread_0:clients=0
```

maybe in the future, we may add more metrics as below

```
# Threads
io_thread_0:clients=0,reads=100,writes=100,used_sys_cpu=3.14,used_user_cpu=0.618
io_thread_1:clients=2,reads=100,writes=100,used_sys_cpu=3.14,used_user_cpu=0.618
io_thread_2:clients=2,reads=100,writes=100,used_sys_cpu=3.14,used_user_cpu=0.618
```

Reply:
LGTM.
thanks.